.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,14 @@
+Thanks for improving cocotb! Here are some points to make this as smooth as possible.
+Not all of them may be applicable.
+
+Most important: please explain *why* you are proposing this change.
+
+* Make sure you have read `CONTRIBUTING.md`.
+* Extend or add a test under `tests/test_cases/`.
+* Add documentation under `documentation/source/`,
+  docstrings in Python code, or Doxygen markup in C/C++ code.
+  Use ``versionadded``/``versionchanged``/``deprecated``.
+* Add a newsfragment - see `documentation/source/newsfragments/README.rst`.
+* Use `closes #XXXX` to auto-close the issue that this PR fixes (if such).
+
+*Please remove this template before submitting.*
```

Comment:
If you surround the template in `<!-- -->`, then there is no need to ask for it to be removed.

Reply:
Done.